src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java

```diff
@@ -57,18 +57,17 @@
    *   There are no other requirements for the object.
    * @param documentID - The optional id for storing the document.
    */
-  public static void storeDocument(Firestore database, String collectionToWriteTo, Object object, String documentID) 
+  public static void storeDocument(Firestore database, String collectionToWriteTo, Object object, Optional<String> documentID) 
       throws FirebaseException {
     // Access the correct collection.
     CollectionReference collectionRef = database.collection(collectionToWriteTo);
 
     // Access/create the new document.
     DocumentReference documentRef;
-    Optional<String> optionalID = Optional.ofNullable(documentID);
-    if (!optionalID.isPresent()) {
+    if (documentID.isEmpty()) {
```

Comment:
Here and below, please follow Optional api. This should be documentId.isPresent().

https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#isPresent--

Reply:
Lol someone told me to change to isEmpty(), I guess this was right originally...